### PR TITLE
RGBW LED dimmer HOMA1001 added

### PR DESCRIPTION
**Pull request by Stefan-3**
Up to now unsupported RGBW LED dimmer HOMA1001 added to shenzhen_homa.js.
This device has two DIP switches (hardware) to configure the connected LED stripe (RGBW, RGB, CT or SC). The new converter supports all 4 configurations. I've tested it.

### CODE REVIEW

**Comment by Koenkk** [comment]
Please add the color temperature range as described on https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_3-adding-converter-s-for-your-device

**Comment by Stefan-3** [comment]
Unfortunetaly I'm not really able to fix the problem:
1) Zigbee-device HOMA1001 paired to Philips Hue bridge supports colour temperature (HOMA1001 in RGBW mode)
2) I'd like to have this in ioBroker too, but if I change the converter file (shenzhen_homa.js) from

extend: extend.light_onoff_brightness_color(),
to
extend: extend.light_onoff_brightness_colortemp_color({colorTempRange: [150, 500]}),

the RGBW control is still working, but I get no ioBorker objekt to control the colour temperature (even if I add the device to the exclude group of the ioBroker zigbee adapter). Personally it's ok for me not to have the colour temperature functionality. It might be a solution to skip the colour temperature functionality of HOMA1001(?).
If I send the payload data to the HOMA1001:
{"read": {"cluster": "lightingColorCtrl", "attributes": ["colorTempPhysicalMin", "colorTempPhysicalMax"]}}
(as described in
https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_3-adding-converter-s-for-your-device
) I get the failure message in the ioBroker log

zigbee.0
2023-05-07 20:36:00.313	warn	ELEVATED publishToState: message received '{"color_mode":"xy"}' from device 00124b0018d5360c type 'HOMA1001'

zigbee.0
2023-05-07 20:36:00.286	warn	ELEVATED publishToState: value generated '69' from device 00124b0018d5360c for 'Link quality'

zigbee.0
2023-05-07 20:36:00.284	warn	ELEVATED publishToState: message received '{"linkquality":69}' from device 00124b0018d5360c type 'HOMA1001'

zigbee.0
2023-05-07 20:36:00.230	warn	Send command to 0x00124b0018d5360c failed with: Code 134 (Unsupported Attribute)

zigbee.0
2023-05-07 20:36:00.106	warn	ELEVATED Change state 'send_payload' at device 0x00124b0018d5360c type 'HOMA1001'

zigbee.0
2023-05-07 20:36:00.085	warn	ELEVATED: User stateChange zigbee.0.00124b0018d5360c.send_payload {"val":"{\n \"read\": {\n \"cluster\": \"lightingColorCtrl\",\n \"attributes\": [\n \"colorTempPhysicalMin\",\n \"colorTempPhysicalMax\"\n ]\n }\n}","ack":false,"ts":1683484560077,"q":0,"from":"system.adapter.admin.0","user":"system.user.admin","lc":1683484560077}

zigbee.0
2023-05-07 20:33:22.920	warn	Send command to 0x00124b0018d5360c failed with: Code 134 (Unsupported Attribute)

So, finally I don't know how to get the min/max value for the colour temperature :-(

**Comment by MattWestb** [comment]
Its one LED drive and its not knowing witch color range the LED stripe is having that is connected then its one external device and its depending of the connected stripes preference.

**Comment by Stefan-3** [comment]
@MattWestb Yes, that's my opinion, too. The HOMA1001 is an LED driver without a specific LED stripe. The colour temperature range depends on the connected LED stripe. From my understanding the temperature range is no capability of the HOMA1001. It's exclusively a capability of the LED stripe connected to the HOMA1001. But if I don't enter a value for the temperature range, the PR (pull request) with its automatic NPM fails because of the missing temperature range.

File: devices/shenzhen_homa.js [hunk 1]
```diff
@@ -3,6 +3,64 @@ const extend = require('../lib/extend');
...
+        extend: extend.light_onoff_brightness_colortemp(),
+    },
+    {
+        fingerprint: [
+            {modelID: 'HOMA1001', endpoints: [
+                {ID: 11, profileID: 49246, deviceID: 256, inputClusters: [0,3,4,5,6,8], outputClusters: []},
+                {ID: 13, profileID: 49246, deviceID: 57694, inputClusters: [4096], outputClusters: [4096]},
+            ]},
+        ],
+        zigbeeModel: ['HOMA1001'],
```
Comment (Koenkk) [inline devices/shenzhen_homa.js:58]: Please remove all `zigbeeModel: ['HOMA1001'],` from the added entries and check if it still work (identification should go via `fingerprint`, not `zigbeeModel`.)

**Comment by Stefan-3** [comment]
Done :-) I've removed the `zigbeeModel: ['HOMA1001'],` entries and the LED driver is still working properly. So we could try again to add it to GIT. @Koenkk can you do this ? (or should I create a new PR? I suppose not)

File: devices/shenzhen_homa.js [hunk 2]
```diff
@@ -3,6 +3,60 @@ const extend = require('../lib/extend');
...
 module.exports = [
+    {
+        fingerprint: [
+            {modelID: 'HOMA1001', endpoints: [
+                {ID: 10, profileID: 49246, deviceID: 256, inputClusters: [0,3,4,5,6,8], outputClusters: []},
+                {ID: 11, profileID: 49246, deviceID: 528, inputClusters: [0,3,4,5,6,8,768], outputClusters: []},
+                {ID: 13, profileID: 49246, deviceID: 57694, inputClusters: [4096], outputClusters: [4096]},
+            ]},
+        ],
+        model: 'HOMA1001',
```
Comment (Koenkk) [inline devices/shenzhen_homa.js:14]: The model should be unique, I suggest to postfix all the models with the functionallity, e.g. `HOMA1001_RGB`, `HOMA1001_CT`, etc.

**Comment by Stefan-3** [comment]
I would have a picture of the device. Should I add this jpg to [https://github.com/Koenkk/zigbee2mqtt.io/tree/master/public/images/devices](url) ? If yes: what file name ? HOMA1001.jpg? HOMA1001_RGBW.jpg? HOMA1001_RGB.jpg? HOMA1001_CT.jpg? Who makes the mapping which jpg (e.g. HOMA1001.jpg) belongs to which zigbee device (e.g. HOMA1001)? Are you doing this manually? I'm missing this information in the documentation [https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_3-adding-converter-s-for-your-device](url)

**Comment by Stefan-3** [comment]
Great. So finally we solved it :-) Many thanks for your efforts and the fantastic support.

**Pull request merged**